### PR TITLE
Feat(eos_cli_config_gen): Add support for `mac timestamp header` command

### LINKED ISSUE #4331: (eos_cli_config_gen) Add mac_timestamp_header to the ethernet_interfaces data model
### Enhancement summary

`mac timestamp header` is a flag you can enable on an interface in EOS. This request is to add the support for this command to `eos_cli_config_gen`

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Whenever we need to enable `mac timestamp header` at the ethernet interface level.

### Describe the solution you would like

Consider this example:
```
ethernet_interfaces:
  - name: <str>
    mac_timestamp_header: <bool>
```
Would result in:
```
interface <str>
    mac timestamp header
```
Code example:
```
{%     if ethernet_interface.mac_timestamp_header is arista.avd.defined(true) %}
   mac timestamp header
{%     endif %}
```

### Describe alternatives you have considered

_No response_

### Additional context

I can see there are already similar flags defined for `mac_access_group_out` and `mac_access_group_in` under `ethernet_interfaces`.
It makes sense to follow this approach with the `mac_timestamp_header` flag.

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by laxmikantchintakindi**
## Change Summary

Add support for `mac timestamp header` command.

## Related Issue(s)

Fixes #4331

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

mac timestamp header is a flag you can enable on an interface in EOS. 
```
interface <str>
    mac timestamp header
```
 
### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by MaheshGSLAB — APPROVED** [review]

**Comment by alexeygorbunov** [comment]
TAP Aggregation tool ports seem to support 3 different `timestamp`-ing modes (depending on a platform):
- `header` - on platforms like `7280E/R`s and `7500E/R`s
- `before-fcs` & `replace-fcs` on platforms like `7150`s

EOS CLI (checked on 4.32.1F) supports all three under interface configuration mode:
```IntfConfigMode: [no|default] mac timestamp ( before-fcs | replace-fcs | header )```

```
  before-fcs   insert timestamp before fcs field
  replace-fcs  replace fcs field with timestamp
  header    insert timestamp in Ethernet header
```

Would it make sense to cover all three available options?

```yaml
  type: dict
  keys:
    ethernet_interfaces:
      items:
        keys:
+        mac_timestamp:
+          type: str
+          valid_values: ["before-fcs", "replace-fcs", "header"]
```


P.S. Although interface-level `mac timestamp <str>` commands are only active on tool port of the EoS switch running in `TAP aggregation mode` - EoS allows to configure 'mac timestamp <str>' event if these conditions are not met, so probably we can skip interface mode validations within schema and can let CV/EOS to highlight warnings (based on workspace build validations) if platform|port_mode|etc is not matching.

**Comment by laxmikantchintakindi** [comment]
> TAP Aggregation tool ports seem to support 3 different `timestamp`-ing modes (depending on a platform):
> 
> * `header` - on platforms like `7280E/R`s and `7500E/R`s
> * `before-fcs` & `replace-fcs` on platforms like `7150`s
> 
> EOS CLI (checked on 4.32.1F) supports all three under interface configuration mode: `IntfConfigMode: [no|default] mac timestamp ( before-fcs | replace-fcs | header )`
> 
> ```
>   before-fcs   insert timestamp before fcs field
>   replace-fcs  replace fcs field with timestamp
>   header    insert timestamp in Ethernet header
> ```
> 
> Would it make sense to cover all three available options?
> 
> ```yaml
>   type: dict
>   keys:
>     ethernet_interfaces:
>       items:
>         keys:
> +        mac_timestamp:
> +          type: str
> +          valid_values: ["before-fcs", "replace-fcs", "header"]
> ```
> 
> P.S. Although interface-level `mac timestamp <str>` commands are only active on tool port of the EoS switch running in `TAP aggregation mode` - EoS allows to configure 'mac timestamp ' event if these conditions are not met, so probably we can skip interface mode validations within schema and can let CV/EOS to highlight warnings (based on workspace build validations) if platform|port_mode|etc is not matching.

Fixed. Thanks.

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/ethernet_interfaces.schema.yml [hunk 1]
```diff
@@ -116,6 +116,13 @@ keys:
             forwarding_profile:
               type: str
               description: L2 protocol forwarding profile.
+        mac_timestamp:
+          type: str
+          valid_values: ["before-fcs", "replace-fcs", "header"]
+          description: |-
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/ethernet_interfaces.schema.yml:122]: please add in the description that the options are platform dependent

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/address-locking.yml [hunk 2]
```diff
@@ -22,19 +22,20 @@ ethernet_interfaces:
     description: Address Locking Interface Testing 1
     address_locking:
       ipv4: true
-
+    mac_timestamp: header
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/address-locking.yml]: why is it in address-locking file? please do it in the ethernet interface one

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/ethernet_interfaces.schema.yml (hunk 1 above)
Comment (laxmikantchintakindi) [reply python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/ethernet_interfaces.schema.yml:122]: Added.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/address-locking.yml (hunk 2 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/address-locking.yml]: Fixed.

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2 [hunk 3]
```diff
@@ -302,6 +302,15 @@ interface {{ ethernet_interface.name }}
...
 {%     endif %}
+{%     if ethernet_interface.mac_timestamp is arista.avd.defined %}
+{%         if ethernet_interface.mac_timestamp == "header" %}
+   mac timestamp header
+{%         elif ethernet_interface.mac_timestamp == "before-fcs" %}
+   mac timestamp before-fcs
+{%         elif ethernet_interface.mac_timestamp == "replace-fcs" %}
+   mac timestamp replace-fcs
+{%         endif %}
+{%     endif %}
```
Comment (alexeygorbunov) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2]: As value is already validated by the schema - can we just render it right away without additional checks?
```suggestion
{%     if ethernet_interface.mac_timestamp is arista.avd.defined %}
   mac timestamp {{ ethernet_interface.mac_timestamp }}
{%     endif %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/ethernet-interfaces.md [hunk 4]
```diff
@@ -26,6 +26,7 @@
     | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;l2_protocol</samp>](## "ethernet_interfaces.[].l2_protocol") | Dictionary |  |  |  |  |
     | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;encapsulation_dot1q_vlan</samp>](## "ethernet_interfaces.[].l2_protocol.encapsulation_dot1q_vlan") | Integer |  |  |  | Vlan tag to configure on sub-interface. |
     | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;forwarding_profile</samp>](## "ethernet_interfaces.[].l2_protocol.forwarding_profile") | String |  |  |  | L2 protocol forwarding profile. |
+    | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;mac_timestamp</samp>](## "ethernet_interfaces.[].mac_timestamp") | String |  |  | Valid Values:<br>- <code>before-fcs</code><br>- <code>replace-fcs</code><br>- <code>header</code> | header: Insert timestamp in ethernet header. Supports on platforms like 7500E/R and 7280E/R.<br>before-fcs: Insert timestamp before fcs field. Support on platforms like 7150.<br>replace-fcs: Replace fcs field with timestamp. |
```
Comment (alexeygorbunov) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/ethernet-interfaces.md]: Potential cosmet. changes
```suggestion
    | [<samp>&nbsp;&nbsp;&nbsp;&nbsp;mac_timestamp</samp>](## "ethernet_interfaces.[].mac_timestamp") | String |  |  | Valid Values:<br>- <code>before-fcs</code><br>- <code>replace-fcs</code><br>- <code>header</code> | header: Insert timestamp in ethernet header. Supported on platforms like 7500E/R and 7280E/R.<br>before-fcs: Insert timestamp before fcs field. Supported on platforms like 7150.<br>replace-fcs: Replace fcs field with timestamp. |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/ethernet-interfaces.md [hunk 5]
```diff
@@ -550,6 +551,11 @@
 
           # L2 protocol forwarding profile.
           forwarding_profile: <str>
+
+        # header: Insert timestamp in ethernet header. Supports on platforms like 7500E/R and 7280E/R.
+        # before-fcs: Insert timestamp before fcs field. Support on platforms like 7150.
```
Comment (alexeygorbunov) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/ethernet-interfaces.md]: Potential cosmet. changes
```suggestion
        # header: Insert timestamp in ethernet header. Supported on platforms like 7500E/R and 7280E/R.
        # before-fcs: Insert timestamp before fcs field. Supported on platforms like 7150.
```

Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/ethernet-interfaces.md]: Fixed.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/ethernet-interfaces.md (hunk 4 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/ethernet-interfaces.md]: Fixed.

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2 (hunk 3 above)
Comment (laxmikantchintakindi) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2]: Fixed.

**Review by alexeygorbunov — APPROVED** [review]
LGTM

**Review by gmuloc — APPROVED** [review]

**Pull request merged**